### PR TITLE
feat(udf): Add deno as UDF language

### PR DESCRIPTION
**Pull request by bakjos**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Second Attempt

This PR add the option to use deno as udf language, it includes `fetch`, `web` and `crypto` API support. 

```sql
CREATE FUNCTION digest( t string ) RETURNS bytea LANGUAGE javascript RUNTIME deno AS $$
    const subtle = crypto.subtle; 
    const key = await subtle.generateKey({
        name: 'HMAC',
        hash: 'SHA-256',
        length: 256,
        }, true, ['sign', 'verify']);
    
        const enc = new TextEncoder();
        const message = enc.encode(t);
    
        const result = await subtle.sign({
        name: 'HMAC',
        }, key, message);
        console.log('result', result);
        return result;
$$ ASYNC;
```

```sql
CREATE FUNCTION delay_response()
    RETURNS TABLE (x int) LANGUAGE javascript RUNTIME deno AS $$
                 const delayedResponses = {
                    delays: [50, 10, 15], 
                    wait(delay) {
                      return new Promise((resolve) => {
                        setTimeout(resolve, delay);
                      });
                    },
                  
                    async *[Symbol.asyncIterator]() {
                      for (const delay of this.delays) {
                        await this.wait(delay);
                        yield delay;
                      }
                    },
                  };
                return delayedResponses;
$$   SYNC;
```

```sql
CREATE FUNCTION fetch_api() RETURNS TABLE ( data struct< id int, name string, gender string >) LANGUAGE javascript RUNTIME deno AS $$
    const response = await fetch('https://rickandmortyapi.com/api/character');
    const resp = await response.json();
    for (const r of resp.results) {
        yield r;
    }
$$ ASYNC GENERATOR;
```

I added support for setting the function type when creating a function 

| type | function | declaration |
| ---- |  ----- | ------ |
| `ASYNC` | A java script async function, to enable the usage of the await api | ` async function <name> { <body> } ` | 
| `ASYNC GENERATOR` | A java script async generator function, to enable the usage of the await and yield api | ` async function* <name> { <body> } ` | 
| `GENERATOR` | A java script generator function, to enable the usage of the yield api | ` function* <name> { <body> } ` | 
| `SYNC` | A regular javascript function, if a promise is returned, will wait until is fulfilled to return the result | `function <name> { <body> } ` | 

If not defined will use `normal` for UDFs and `generator` for table functions;

I put the deno language support under a feature flag, which is currently enabled by default. (Plan to remove before this PR gets merged).

It will require https://github.com/risingwavelabs/arrow-udf/pull/7 to get merged 

I'd also like to use one of the table functions as a source (to receive data from websockets or sse). But it's not currently supported. What do you think would be a good way to do it?

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [x] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [x] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by bakjos** [comment]
@wangrunji0408  I think I've finally found the root cause of the flaky test, this PR explains it https://github.com/risingwavelabs/arrow-udf/pull/13 

This build was running the function 10 times <https://buildkite.com/risingwavelabs/pull-request/builds/47231#018ecf5a-feee-4897-98f4-a68f9de54fd0>

And this one only once  https://buildkite.com/risingwavelabs/pull-request/builds/47232#018ecf76-e420-4f4e-8bb7-1bedd62fe6ba

**Review by wangrunji0408 — APPROVED** [review]
Thank you!

**Comment by wangrunji0408** [comment]
kindly request a quick review of cargo.lock @xiangjinwu @BugenZhao

File: src/meta/dashboard/Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ axum = { workspace = true }
 axum-embed = "0.1"
 bytes = "1"
 mime_guess = "2"
-reqwest = "0.12.2"
+reqwest = "0.12"
```
Comment (xiangjinwu) [inline src/meta/dashboard/Cargo.toml:17]: Why do we need to allow 0.12.0 / 0.12.1 here?

**Review by xiangjinwu — COMMENTED** [review]
With the help of a script I compared this `Cargo.lock` with its merge-base (02a3985548bfa41cf8d7aa93235c2be76efe0233):
```
lhs: 1207
rhs: 1322
----
same 1192
diff 15
add 115
del 0
----
DN aes: 0.8.4 -> 0.8.3
UP curve25519-dalek: 4.1.0 -> 4.1.1
UP ecdsa: 0.16.8 -> 0.16.9
UP ed25519-dalek: 2.0.0 -> 2.1.1
UP elliptic-curve: 0.13.6 -> 0.13.8
DN hyper: 1.2.0 -> 1.1.0
UP miniz_oxide: 0.7.1 -> 0.7.2
UP once_cell: 1.18.0 -> 1.19.0
UP os_pipe: 1.1.4 -> 1.1.5
UP primeorder: 0.13.2 -> 0.13.6
UP rsa: 0.9.2 -> 0.9.5
UP sha1: 0.10.5 -> 0.10.6
UP sha2: 0.10.7 -> 0.10.8
UP signature: 2.0.0 -> 2.2.0
UP triomphe: 0.1.9 -> 0.1.11
```
* The comparison of [Package](https://docs.rs/cargo-lock/latest/cargo_lock/package/struct.Package.html) ignored dependencies. For example `foo 1 -> bar 1.2` changing to `foo 1 -> bar 1.1` will still consider `foo 1` as `same`.
* The downgrade of `aes` and `hyper` were due to [`deno_crypto`](https://crates.io/crates/deno_crypto/0.161.0/dependencies) and [`deno_http`](https://crates.io/crates/deno_http/0.144.0/dependencies) respectively.
* Feels like this script could be a bot - to post `Cargo.lock` upgrades and downgrades to relevant PR. Anyone happens to know an existing one?

File: src/meta/dashboard/Cargo.toml (hunk 1 above)
Comment (bakjos) [reply src/meta/dashboard/Cargo.toml]: A leftover from the previous version, I rolled back this change

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**